reconcile/utils/ocm.py

```diff
@@ -508,6 +509,64 @@ class OCMServiceAccountNotAssociatedToOrg(Exception):
     pass
 
 
+class SectorConfigError(Exception):
+    pass
+
+
+# A weak reference to a sector. Used to reference a sector dependency, even if this one is
+# not available yet
+@dataclass
+class SectorWeakReference:
```

Comment:
a lot of the sector deps graph related code could be streamlined by using a graph library like networkx. for change-type inheritance (https://github.com/app-sre/qontract-reconcile/blob/fa3b25276da70348e5faa2d36fec84077ad1a92b/reconcile/change_owners/change_types.py#L408) it solved the weak reference, cycle detection and path traversal with a tiny amount of code.

no need to change anything here. just an idea for future changes whenever you need to tackle more challenging scenarios with sector interdependencies.

Reply:
I'll have a look, thanks for the review!